### PR TITLE
Add a basic Long and Double gauge implementation 

### PR DESCRIPTION
**Pull request by thisthat**
This addresses #684 and it is implemented following #755

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleGauge.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleGauge.java]: 2020

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleGauge.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  public int hashCode() {
+    int result = super.hashCode();
+    result = 31 * result + (monotonic ? 1 : 0);
+    return result;
+  }
+
+  private static class SdkBoundDoubleGauge extends BaseBoundInstrument implements BoundDoubleGauge {
+
+    private final boolean monotonic;
+    @Nullable private Double prev = null;
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleGauge.java]: let's not abbreviate this. Just call it "currentValue"

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/SdkLongGauge.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  public int hashCode() {
+    int result = super.hashCode();
+    result = 31 * result + (monotonic ? 1 : 0);
+    return result;
+  }
+
+  private static class SdkBoundLongGauge extends BaseBoundInstrument implements BoundLongGauge {
+
+    private final boolean monotonic;
+    @Nullable private Long prev = null;
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/SdkLongGauge.java]: same comments on naming apply here

**Comment by bogdandrutu** [comment]
Thank you for this PR @thisthat, I will hold on merging this PR until we discuss https://github.com/open-telemetry/opentelemetry-specification/issues/412. If you think we should make progress we can merge it and maybe remove later if the decision will be to remove the Gauge instrument.

**Comment by thisthat** [comment]
I would argue to hold this PR until a decision at the specification level is taken. If the Gauge API will change/removed, I will update/delete this PR accordingly.

**Comment by bogdandrutu** [comment]
Based on the https://github.com/open-telemetry/oteps/pull/80, I will close this PR, feel free to reopen if needed.

**Pull request closed without merging**